pkg/combustion/fips.go

```diff
@@ -0,0 +1,49 @@
+package combustion
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"github.com/suse-edge/edge-image-builder/pkg/log"
+)
+
+const (
+	fipsComponentName = "fips"
+	fipsScriptName    = "15-fips-setup.sh"
+)
+
+//go:embed templates/15-fips-setup.sh
+var fipsScript string
+
+func configureFips(ctx *image.Context) ([]string, error) {
+	fips := ctx.ImageDefinition.OperatingSystem.Fips
+	if !fips {
+		log.AuditComponentSkipped(fipsComponentName)
+		return nil, nil
+	}
+
+	if err := writeFipsCombustionScript(ctx); err != nil {
+		log.AuditComponentFailed(fipsComponentName)
+		return nil, err
+	}
+
+	kernelArgList := ctx.ImageDefinition.OperatingSystem.KernelArgs
+	kernelArgList = append(kernelArgList, "fips=1")
+	ctx.ImageDefinition.OperatingSystem.KernelArgs = kernelArgList
```

Comment:
Modifications to the input context shouldn't happen here. I'd suggest to:
- rename `appendFipsRPMs()` to `appendFIPS()` in the `eib` package
- extract an `appendKernelArgs()` function accepting variadic string args
- call `appendKernelArgs()` from `appendFIPS()`